src/ape/api/config.py

```diff
@@ -93,3 +122,330 @@ class GenericConfig(ConfigDict):
+
+        return result
+
+    def write_to_disk(self, destination: Path, replace: bool = False):
+        """
+        Write this config to a file.
+
+        Args:
+            destination (Path): The path to write to.
+            replace (bool): Set to ``True`` to overwrite the file if it exists.
+        """
+        if destination.exists() and not replace:
+            raise ValueError(f"Destination {destination} exists.")
+        elif replace:
+            destination.unlink(missing_ok=True)
+
+        if destination.suffix in (".yml", ".yaml"):
+            destination.parent.mkdir(parents=True, exist_ok=True)
+            with open(destination, "x") as file:
+                data = self.model_dump(by_alias=True, mode="json")
+                yaml.safe_dump(data, file)
+
+        elif destination.suffix == ".json":
+            destination.write_text(self.model_dump_json(by_alias=True))
+
+        else:
+            raise ValueError(f"Unsupported destination file type {destination}.")
+
+
+def _get_compile_configs_from_manifest(manifest: PackageManifest) -> dict[str, dict]:
+    configs: dict[str, dict] = {}
+    for compiler in [x for x in manifest.compilers or [] if x.settings]:
+        name = compiler.name.strip().lower()
+        compiler_data = {}
+        settings = compiler.settings or {}
+        remapping_list = []
+        for remapping in settings.get("remappings") or []:
+            parts = remapping.split("=")
+            key = parts[0]
+            link = parts[1]
+            if link.startswith(f".cache{os.path.sep}"):
+                link = os.path.sep.join(link.split(f".cache{os.path.sep}"))[1:]
+
+            new_entry = f"{key}={link}"
+            remapping_list.append(new_entry)
+
+        if remapping_list:
+            compiler_data["import_remapping"] = remapping_list
+
+        if "evm_version" in settings:
+            compiler_data["evm_version"] = settings["evm_version"]
+
+        if compiler_data:
+            configs[name] = compiler_data
+
+    return configs
+
+
+def _get_dependency_configs_from_manifest(manifest: PackageManifest) -> dict:
+    dependencies_config: list[dict] = []
+    dependencies = manifest.dependencies or {}
+    for package_name, uri in dependencies.items():
+        if "://" not in str(uri) and hasattr(uri, "scheme"):
+            uri_str = f"{uri.scheme}://{uri}"
+        else:
+            uri_str = str(uri)
+
+        dependency: dict = {"name": str(package_name)}
+        if uri_str.startswith("https://"):
+            # Assume GitHub dependency
+            version = uri_str.split("/")[-1]
+            dependency["github"] = uri_str.replace(f"/releases/tag/{version}", "")
+            dependency["github"] = dependency["github"].replace("https://github.com/", "")
+
+            # NOTE: If version fails, the dependency system will automatically try `ref`.
+            dependency["version"] = version
+
+        elif uri_str.startswith("file://"):
```

Comment:
probably a:

```python
        else:
            logger.error(f"Manifest URI {uri_str} not a supported dependency.")
            continue
```

so that ape can still attempt to do its best with weird manifests it may come across